Mathlib/Tactic/Linter/DirectoryDependency.lean

```diff
@@ -117,11 +133,70 @@ def findAny (r : NamePrefixRel) (n₁ : Name) (ns : Array Name) : Option (Name 
         pure ()
     none
 
+/-- Does `r` contain any entries with key `n`? -/
+def containsKey (r : NamePrefixRel) (n : Name) : Bool := NameMap.contains r n
+
 /-- Is a prefix of `n₁` related to a prefix of `n₂`? -/
 def contains (r : NamePrefixRel) (n₁ n₂ : Name) : Bool := (r.find n₁ n₂).isSome
 
 end NamePrefixRel
 
+-- TODO: add/extend tests for this linter, to ensure the allow-list works
```

Comment:
Do you think you can add at least one test so we can see the message format reflected in code? For example we could create a new directory in `MathlibTest` and add it to the `allowedImportDirs`.

Reply:
Good idea. Finding out how to make this work took me a while, but I think I got something working.